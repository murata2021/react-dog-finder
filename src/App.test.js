import { render, screen,fireEvent } from '@testing-library/react';
import App from './App';
import {MemoryRouter} from 'react-router-dom'

it('renders without crashing', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

it('renders with correct elements',()=>{

  const {debug,getByText,queryByText}=render(<MemoryRouter><App/></MemoryRouter>)

  debug()
  let link=getByText('Whiskey')
  expect(getByText('Whiskey loves eating popcorn.', {exact: false})).toBeInTheDocument()
  expect(getByText('Duke believes that ball is life.', {exact: false})).toBeInTheDocument()

  fireEvent.click(link)
  expect(getByText('Whiskey loves eating popcorn', {exact: false})).toBeInTheDocument()
  expect(queryByText('Duke believes that ball is life.', {exact: false})).not.toBeInTheDocument()
  debug()
  
})

it('it redirects to home page when invalid url is given',()=>{

  const {debug,getByText,queryByText}=render(<MemoryRouter initialEntries={['/dogs/blargh']}><App/></MemoryRouter>)
  expect(getByText('Whiskey loves eating popcorn.', {exact: false})).toBeInTheDocument()
  expect(getByText('Duke believes that ball is life.', {exact: false})).toBeInTheDocument()
})

it('redirects when the go to home button is clicked',()=>{

  const {debug,getByText,queryByText}=render(<MemoryRouter initialEntries={['/dogs/Whiskey']}><App/></MemoryRouter>)
  expect(getByText('Whiskey loves eating popcorn.', {exact: false})).toBeInTheDocument()
  expect(queryByText('Duke believes that ball is life.', {exact: false})).not.toBeInTheDocument()
  let button=getByText('Go to main page!')

  fireEvent.click(button)
  expect(getByText('Whiskey loves eating popcorn.', {exact: false})).toBeInTheDocument()
  expect(queryByText('Duke believes that ball is life.', {exact: false})).toBeInTheDocument()
  expect(queryByText('Go to main page!')).not.toBeInTheDocument()
})