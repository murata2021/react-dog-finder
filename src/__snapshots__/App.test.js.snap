// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="Nav"
    >
      <a
        href="/dogs/Whiskey"
      >
        Whiskey
      </a>
      <a
        href="/dogs/Duke"
      >
        Duke
      </a>
      <a
        href="/dogs/Perry"
      >
        Perry
      </a>
      <a
        href="/dogs/Tubby"
      >
        Tubby
      </a>
    </nav>
    <div
      class="container"
    >
      <h1>
        HELLOOOOOOO
      </h1>
      <div
        class="row"
      >
        <div
          class="col-3 mt-4"
        >
          <div
            class="card"
          >
            <img
              alt=""
              class="card-img-top"
              src="whiskey.jpg"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Whiskey (5)
              </h5>
              <p
                class="card-text"
              >
                Whiskey loves eating popcorn.Whiskey is a terrible guard dog.Whiskey wants to cuddle with you!
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-3 mt-4"
        >
          <div
            class="card"
          >
            <img
              alt=""
              class="card-img-top"
              src="duke.jpg"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Duke (3)
              </h5>
              <p
                class="card-text"
              >
                Duke believes that ball is life.Duke likes snow.Duke enjoys pawing other dogs.
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-3 mt-4"
        >
          <div
            class="card"
          >
            <img
              alt=""
              class="card-img-top"
              src="perry.jpg"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Perry (4)
              </h5>
              <p
                class="card-text"
              >
                Perry loves all humans.Perry demolishes all snacks.Perry hates the rain.
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-3 mt-4"
        >
          <div
            class="card"
          >
            <img
              alt=""
              class="card-img-top"
              src="tubby.jpg"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Tubby (4)
              </h5>
              <p
                class="card-text"
              >
                Tubby is really stupid.Tubby does not like walks.Angelina used to hate Tubby, but claims not to anymore.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
